Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polyline_snapping.h

```diff
@@ -0,0 +1,324 @@
+  typedef typename Input_kernel::Line_3 Inexact_line_3;
+  typedef typename Input_kernel::Segment_3 Inexact_segment_3;
+  typedef typename Exact_kernel::Point_3 Point_3;
+  typedef typename Exact_kernel::Line_3 Line_3;
+  typedef typename Exact_kernel::Segment_3 Segment_3;
+  
+  Exact_kernel ek;
+  Exact_to_double exact_to_double;
+
+  Point_3 to_exact(const Inexact_point_3& p) const
+  {
+    return Point_3(p.x(), p.y(), p.z());
+  }
+  Inexact_point_3 to_inexact(const Point_3& p, const Exact_to_double& etd) const
+  {
+    return Inexact_point_3(etd(p.x()),
+                           etd(p.y()),
+                           etd(p.z()));
+  }
+
+public:
+
+  Exact_snapping()
+  {
+
+  }
+
+  std::pair<Inexact_point_3, bool> intersection (const Inexact_segment_3& s0, const Inexact_segment_3& s1) const
+  {
+    Segment_3 exact_s0 (to_exact (s0.source()), to_exact (s0.target()));
+    Segment_3 exact_s1 (to_exact (s1.source()), to_exact (s1.target()));
+    Point_3 result;
+    
+    if (exact_snapping<Exact_kernel> (exact_s0, exact_s1, result))
+      return std::make_pair (to_inexact (result, exact_to_double), true);
+
+    return std::make_pair (Inexact_point_3(), false);
+  }
+  
+};
+
+template <class Kernel>
+class Exact_snapping<Kernel,true>
+{
+//typedefs
+  typedef typename Kernel::Point_3 Point_3;
+  typedef typename Kernel::Line_3 Line_3;
+  typedef typename Kernel::Segment_3 Segment_3;
+public:
+
+  Exact_snapping()
+  {
+
+  }
+
+  std::pair<Point_3, bool> intersection (const Segment_3& s0, const Segment_3& s1) const
+  {
+    Point_3 result;
+    bool out = exact_snapping<Kernel> (s0, s1, result);
+    return std::make_pair (result, out);
+  }
+  
+};
+  
+template <typename PolylineRange,
+          typename OutputIterator,
+          typename Kernel>
+struct Intersect_polylines
+{
+  typedef typename Kernel::Point_3 Point_3;
+  typedef typename Kernel::Segment_3 Segment_3;
+  typedef typename Kernel::FT FT;
+  typedef typename CGAL::Box_intersection_d::Box_with_info_d<double, 3, std::pair<std::size_t, std::size_t> > Box;
+  
+  typedef cpp11::tuple<std::size_t, std::size_t, std::size_t, std::size_t, FT, FT> OutputType;
+  
+  const PolylineRange& polylines;
+  double squared_tolerance;
+  Exact_snapping<Kernel> exact_snapping;
+  mutable OutputIterator output;
+
+  Intersect_polylines (const PolylineRange& polylines,
+                       double tolerance,
+                       OutputIterator output)
+    : polylines (polylines)
+    , squared_tolerance (tolerance * tolerance)
+    , output (output)
+  { }
+
+  void operator()(const Box* b, const Box* c) const
+  {
+    Segment_3 s1 (polylines[b->info().first][b->info().second],
+                  polylines[b->info().first][b->info().second + 1]);
+    Segment_3 s2 (polylines[c->info().first][c->info().second],
+                  polylines[c->info().first][c->info().second + 1]);
+    if (CGAL::squared_distance (s1, s2) > squared_tolerance)
```

Comment:
aren't you somehow doing twice the job with `exact_snapping.intersection()`? I mean couldn't you use `new_point` directly?